File: 백준/Silver/17103. 골드바흐 파티션/골드바흐 파티션.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <math.h>

#define LIMIT (1000000)

int main(void) {
    int* arr = (int*)malloc(sizeof(int) * (LIMIT + 1));
    memset(arr, 1, sizeof(int) * (LIMIT + 1));
    arr[0] = 0, arr[1] = 0;
	for (int i = 2; i < (int)sqrt(LIMIT) + 1; i++) {
		int j = 2;
		while (i * j < LIMIT + 1) {
			arr[i * j] = 0;
			j++;
		}
	}

	int n, num;
	scanf("%d", &n);
	while (n--) {
		scanf("%d", &num);
		int cnt = 0;
		for (int i = 2; i < num / 2 + 1; i++) {
			int j = num - i;
			if (arr[i] && arr[j]) cnt++;
		}
		printf("%d\n", cnt);
	}
    free(arr);
    return 0;
}

File: 백준/Silver/17103. 골드바흐 파티션/골드바흐 파티션.py
import math

def goldPart(t, arr):
    hap = 0
    for i in range(2, int(n / 2) + 1):
        j = n - i
        if(arr[i] and arr[j]): hap += 1
    return hap

limit = 1000000
prime = [True] * (limit + 1)
prime[0] = prime[1] = False
for i in range(2, int(math.sqrt(limit)) + 1):
    if (prime[i] == True):
        j = 2
        while i * j <= limit:
            prime[i * j] = False
            j += 1
  
t = int(input())
while(t > 0):
    n = int(input())
    print(goldPart(t, prime))
    t -= 1